packages/dev/core/src/Physics/v2/physicsBody.ts

```diff
@@ -0,0 +1,170 @@
+import type { IPhysicsEnginePluginV2, MassProperties } from "./IPhysicsEnginePluginV2";
+import type { PhysicsShape } from "./physicsShape";
+import { Vector3 } from "../../Maths/math.vector";
+import type { Scene } from "../../scene";
+
+/**
+ *
+ */
+export class PhysicsBody {
+    /** @internal */
+    public _pluginData: any = undefined;
+
+    private _physicsPlugin: IPhysicsEnginePluginV2 | undefined;
+
+    /**
+     *
+     * @param scene
+     * @returns
+     */
+    constructor(scene: Scene) {
+        if (!scene) {
+            return;
+        }
+        this._physicsPlugin = scene.getPhysicsEngine()?.getPhysicsPlugin() as any;
+        this._physicsPlugin?.initBody(this);
+    }
+    /**
+     *
+     * @param shape
+     */
+    public setShape(shape: PhysicsShape): void {

```

Comment:
what I mean with bound proxy - this function is actually a bound variant of this._physicsPlugin?.setShape(this, shape), forcing the first variable to be "this", which is the same with most of the other functions.